### PR TITLE
Add more tests in: mocking console during testing, and add missing test cases for web hook modules

### PR DESCRIPTION
**Pull request by samhwang**
## Description

This PR will add in a couple of changes:

- Add a `jest-mock-console` package, and add tests that cover the specific console line in all of the modules.
- Refactor embedLink, animatedEmoji and Poll modules and their tests, since they are missing a lot of the logic in their test cases. (Refer to the next section)

## Motivation and Context

This is mainly based on the fact that we're not getting full coverage in our test cases just because of console log lines. During the course of working on this PR, I came in to realize that the problem was much more severe when we are actually missing some test case logic as well, and all of them was not reported by Jest/Istanbul code coverage tool, and only shows up after I started mocking the console module.

## How Has This Been Tested?

This test has been tested (again) by Jest/Istanbul to ensure we have 100% coverage, both locally and via our CI/CD process.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: discord-bot/src/commands/embedLink/index.test.ts [hunk 1]
```diff
@@ -18,17 +19,121 @@ const mockedMessageFetch = fetchMessageObjectById as jest.MockedFunction<
...
+        fetchWebhooks: async () => fakeWebhooks,
+        createWebhook: async () => fakeHook,
+        id: '123',
+      },
+      delete: msgDeleteMock,
+    };
+    await embedLink(mockMsg);
+    expect(webhookSendMock).not.toHaveBeenCalled();
+    expect(msgDeleteMock).not.toHaveBeenCalled();
+  });
```
Comment (jkiller295) [inline discord-bot/src/commands/embedLink/index.test.ts:40]: Cái này gặp `author.bot` nó return mất tiêu rồi

Comment (samhwang) [reply discord-bot/src/commands/embedLink/index.test.ts:40]: oops

File: discord-bot/src/commands/embedLink/index.ts [hunk 2]
```diff
@@ -44,8 +43,6 @@ const embedLink = async (msg: Message) => {
 
   const [firstUrl] = hasDiscordUrl;
   const idString = firstUrl.replace('https://discord.com/channels/', '');
-  if (idString.trim().length === 0 || idString.split('/').length < 3) return; // return if link is wrong
-
```
Comment (jkiller295) [inline discord-bot/src/commands/embedLink/index.ts:48]: Sao không check cái này? Lỡ ai cố tình pen-test thì sao?

Comment (samhwang) [reply discord-bot/src/commands/embedLink/index.ts:48]: vì cái regex ở trên nó cover rồi

cái idString.trim().length === 0 thì để em coi lại, chứ cái idString.split('/').length mà nó < 3 thì nó k thể pass cái regex ở trên được

**Review by jkiller295 — CHANGES_REQUESTED** [review]
Đọc comment pls :D

**Review by dohaicuong — APPROVED** [review]

**Pull request merged**